src/QirRuntime/lib/QAPI/QAPI.cpp

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <assert.h>
+#include <vector>
+
+#include "QAPI.hpp"
+
+#include "IQuantumApi.hpp"
+#include "SimFactory.hpp"
+#include "__quantum__rt.hpp"
+
+using namespace std;
+using namespace Microsoft::Quantum;
+
+static IQuantumApi* GetQuantumExecutor(QuantumExecutionContext* context)
+{
+    return reinterpret_cast<IQuantumApi*>(context->hQuantumApi);
+}
+
+QuantumExecutionContext* QAPI_CreateContext_Toffoli() // NOLINT
+{
+    unique_ptr<IQuantumApi> qapi = CreateToffoliSimulator();

```

Comment:
The factory method returns a unique_ptr, which is a better (imho) memory management choice for internal clients who will get all the benefits of automatic memory management it provides. The QuantumExecutionContext hasn't been thought through (copied from some preexisting experimental code) and is mostly just a stub to indicate that we might need to manage more environment data than a single simulator while exposing a primitive C-API for testing the scenario of consuming the runtime via dll rather than statically.